rust/lance/src/dataset/refs.rs

```diff
@@ -216,43 +347,345 @@ impl Tags {
+                )
+                .await?;
+                Ok((name, contents))
+            })
+            .buffer_unordered(10)
+            .try_collect()
+            .await
+    }
+
+    pub async fn list(&self) -> Result<HashMap<String, BranchContents>> {
+        self.fetch()
+            .await
+            .map(|branches| branches.into_iter().collect())
+    }
+
+    pub async fn get(&self, branch: &str) -> Result<BranchContents> {
+        check_valid_branch(branch)?;
+
+        let root_location = self.refs.root()?;
+        let branch_file = branch_contents_path(&root_location.path, branch);
+
+        if !self.object_store().exists(&branch_file).await? {
+            return Err(Error::RefNotFound {
+                message: format!("branch {} does not exist", branch),
+            });
+        }
+
+        let branch_contents = BranchContents::from_path(&branch_file, self.object_store()).await?;
+
+        Ok(branch_contents)
+    }
+
+    pub async fn create(
+        &mut self,
+        branch_name: &str,
+        version_number: u64,
+        source_branch: Option<&str>,
+    ) -> Result<()> {
+        check_valid_branch(branch_name)?;
+
+        let source_branch = source_branch.map(String::from);
+        let root_location = self.refs.root()?;
+        let branch_file = branch_contents_path(&root_location.path, branch_name);
+        if self.object_store().exists(&branch_file).await? {
+            return Err(Error::RefConflict {
+                message: format!("branch {} already exists", branch_name),
+            });
+        }
+
+        let branch_location = self.refs.base_location.find_branch(source_branch.clone())?;
+        // Verify the source version exists
+        let manifest_file = self
+            .refs
+            .commit_handler
+            .resolve_version_location(
+                &branch_location.path,
+                version_number,
+                &self.refs.object_store.inner,
+            )
+            .await?;
+
+        if !self.object_store().exists(&manifest_file.path).await? {
+            return Err(Error::VersionNotFound {
+                message: format!("Manifest file {} does not exist", &manifest_file.path),
+            });
+        };
+
+        let branch_contents = BranchContents {
+            parent_branch: source_branch,
+            parent_version: version_number,
+            create_at: chrono::Utc::now().timestamp() as u64,
+            manifest_size: if let Some(size) = manifest_file.size {
+                size as usize
+            } else {
+                self.object_store().size(&manifest_file.path).await? as usize
+            },
+        };
+
+        self.object_store()
+            .put(
+                &branch_file,
+                serde_json::to_string_pretty(&branch_contents)?.as_bytes(),
+            )
+            .await
+            .map(|_| ())
+    }
+
+    pub async fn delete(&mut self, branch: &str, forced: bool) -> Result<()> {
```

Comment:
should add docstring explain the meaining of force. Also I think typically should use `force` instead of `forced` for the variable name.